src/webviews/webview-side/ipywidgets/common/scriptManager.ts

```diff
@@ -199,6 +201,7 @@ export class ScriptManager extends EventEmitter {
         // Now resolve promises (anything that was waiting for modules to get registered can carry on).
         sources.forEach((source) => {
             this.registeredWidgetSources.set(source.moduleName, source);
+            this.widgetsRegisteredInRequireJs.add(source.moduleName);
```

Comment:
Not sure how this went missing, but this was also required. Even if this wasn't added, then we'd re-download the js file and try all over again, this is meant to avoid that.